lib/vicadmin/validate.go

```diff
@@ -191,3 +202,55 @@ func (v *Validator) QueryDatastore(ctx context.Context, vch *config.VirtualConta
 			</div>`, v.StorageRemaining, ds.Name, float64(ds.Summary.FreeSpace)/(1<<30)))
 	}
 }
+
+func (v *Validator) QueryVCHStatus(vch *config.VirtualContainerHostConfigSpec) {
+	defer trace.End(trace.Begin(""))
+	v.VCHIssues = template.HTML("")
+	v.VCHStatus = GoodStatus
+
+	procs := []string { "vic-init" }
+
+	// Extract required components from vchConfig
+	// Only report on components with Restart set to true
+	for _, sess := range vch.ExecutorConfig.Sessions {
+		if !sess.Restart {
+			continue
+		}
+		parsedcmd := basename.FindStringSubmatch(sess.Cmd.Path)
```

Comment:
Maybe you need to write a helper function that return the pid file path given a session object. That way you can consistently use it everywhere in the tether and vidadmin.